### PR TITLE
feat: Update reference/CRDs.md for matchConditions

### PR DESCRIPTION
**Pull request by viccuad**
## Description

Fix https://github.com/kubewarden/kubewarden-controller/issues/758

File taken from
kubewarden-controller/docs/crds/CRD-docs-for-docs-repo.md

## Test

To test this pull request, you can run the following commands:

### CODE REVIEW

File: docs/reference/CRDs.md
Comment (jhkrug) [inline docs/reference/CRDs.md:106]:
```suggestion
| `matchConditions` _[MatchCondition](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.28/#matchcondition-v1-admissionregistration) array_ | MatchConditions are a list of conditions that must be met for a request to be validated. Match conditions filter requests that have already been matched by the rules, namespaceSelector, and objectSelector. An empty list of matchConditions matches all requests. There are a maximum of 64 match conditions allowed. If a parameter object is provided, it can be accessed via the `params` handle in the same manner as validation expressions. The exact matching logic is (in order): 1. If ANY matchCondition evaluates to FALSE, the policy is skipped. 2. If ALL matchConditions evaluate to TRUE, the policy is evaluated. 3. If any matchCondition evaluates to an error (but none are FALSE): - If failurePolicy=Fail, reject the request - If failurePolicy=Ignore, the policy is skipped. Only available if the feature gate AdmissionWebhookMatchConditions is enabled. |
```

File: docs/reference/CRDs.md
Comment (jhkrug) [inline docs/reference/CRDs.md:288]:
```suggestion
| `tolerations` _[Toleration](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.28/#toleration-v1-core) array_ | Tolerations describe the policy server pod's tolerations. It can be used to ensure that the policy server pod is not scheduled onto a node with a taint. |
```

**Review by jhkrug — CHANGES_REQUESTED** [review]
Hi @viccuad. A couple of minor suggestions, I assume they need changing in the kubewarden-controller code. I think it got done in 822 there. Sorry, I missed it. Feel free to use the area/docs tag and/or ping me directly for any strings/docs in the code I should take a look at.

**Comment by viccuad** [comment]
@jhkrug good catches, thanks!

I opened https://github.com/kubewarden/kubewarden-controller/pull/826 that contains those fixes, and also adds a CODEOWNERS on that folder so that github will include @kubewarden/kubewarden-documentation as reviewer in the future.

Once that PR is merged, I will rebase this one with the fixes.

**Comment by viccuad** [comment]
rebased with fixes, please re-review.

**Review by jhkrug — APPROVED** [review]
lgtm. thanks.

**Pull request merged**
